From cb9f6af0949d760d20867c517770ba7b7ef1e624 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 28 May 2014 15:11:59 +0200
Subject: [PATCH] iscsi/parse-iscsiroot.sh: call iscsistart regardless

call iscsistart, even if no network config is given or
- link down temporarily down
- changed MAC address
- replaced/swapped NIC
- misconfigured kernel command line

https://bugzilla.redhat.com/show_bug.cgi?id=1099603
---
 modules.d/95iscsi/parse-iscsiroot.sh | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/modules.d/95iscsi/parse-iscsiroot.sh b/modules.d/95iscsi/parse-iscsiroot.sh
index 939bd2c..0e3169f 100755
--- a/modules.d/95iscsi/parse-iscsiroot.sh
+++ b/modules.d/95iscsi/parse-iscsiroot.sh
@@ -79,10 +79,15 @@ if ! [ -e /sys/module/iscsi_tcp ]; then
     modprobe -q iscsi_tcp || die "iscsiroot requested but kernel/initrd does not support iscsi"
 fi
 
-if [ -n "$netroot" ] && [ "$root" != "/dev/root" ] && [ "$root" != "dhcp" ]; then
-    if ! getarg "ip="; then
-        initqueue --onetime --settled /sbin/iscsiroot dummy "$netroot" "$NEWROOT"
-    fi
+if [ -n "$iscsi_firmware" ] || \
+    ( [ -n "$netroot" ] && [ "$root" != "/dev/root" ] && [ "$root" != "dhcp" ] && ! getarg "ip=" )
+    then
+    initqueue --onetime --settled /sbin/iscsiroot dummy "$netroot" "$NEWROOT"
+else
+    # also call iscsiroot for misconfigured "ip=" or link down or MAC addr change
+    # or replaced NIC
+    # in the timeout queue
+    initqueue --onetime --timeout /sbin/iscsiroot dummy "$netroot" "$NEWROOT"
 fi
 
 netroot_enc=$(str_replace "$netroot" '/' '\2f')
